apps/dashboard/src/components/billing/plan-action-button.tsx

```diff
@@ -0,0 +1,67 @@
+import { Button } from '@/components/primitives/button';
+import { ApiServiceLevelEnum } from '@novu/shared';
+import { useMutation } from '@tanstack/react-query';
+import { get, post } from '../../api/api.client';
+import { toast } from 'sonner';
+import { useSubscription } from './hooks/use-subscription';
+import { cn } from '../../utils/ui';
+import { ChevronRight } from 'lucide-react';
+
+interface PlanActionButtonProps {
+  selectedBillingInterval: 'month' | 'year';
+  variant?: 'default' | 'outline';
+  showIcon?: boolean;
+  className?: string;
+  size?: 'default' | 'sm' | 'lg';
+}
+
+export function PlanActionButton({
+  selectedBillingInterval,
+  variant = 'default',
+  showIcon = false,
+  className,
+  size = 'default',
+}: PlanActionButtonProps) {
+  const { data: subscription } = useSubscription();
+  const { trial, apiServiceLevel } = subscription || {};
+  const isPaidSubscriptionActive =
+    subscription?.isActive && !trial?.isActive && apiServiceLevel !== ApiServiceLevelEnum.FREE;
```

Comment:
If the subscription is null until it loads, we can easily return null at the top of the component (after the hooks) and avoid all the `?.` train wrecks.

Reply:
I will keep this, but will refactor slightly. I want to use the Button loading indication. I think it's looking better.